### PR TITLE
AC2 - Course registration

### PR DESCRIPTION
**Pull request by Divyanshusir**
## AC-2 All the changes made so far in the module : 

JSON APIs created for
Student :
student_pre_registration ,
student_final_registration ,
student_view_registration ,
view_offered_courses ,
student_backlog_courses ,
add_course ,
drop_course,
academic_procedures_student ,

Acad Admin :
acad_view_reigstrations,
get_course_list,
configure_pre_registration_date,
configure_final_registration_date

Faculty :
faculty_assigned_courses

Model created for:
Backlog course

Created templates for :
Doing pre registration configuration for Academic person -( seeing courses available for pre-registration in any semester , and functionality to add and remove any course from the Course slot .)
Showing backlog courses of a student,
Functionality For Faculty to View Assigned Courses and generate Roll List,

Small changes made and bug fixed : 

In FusionIIIT/applications/academic_information/views.py (line 1033) - for solving the error when fetching the excel sheet if pre registration details of any student is not found ,
In FusionIIIT/applications/academic_procedures/views.py (line 415) - corrected the working of get_drop_course options so as student can see the courses in the drop course section
In FusionIIIT/applications/academic_procedures/views.py (line 1845) - modified the function to fix bug in get_batch_course so as to handle the batches which are after 2021 as their ids have different pattern
In FusionIIIT/applications/academic_procedures/views.py (line 2283 , 2288) - modified the function to solve the problem of repeated verification  of same student 

https://github.com/FusionIIIT/Fusion/assets/114430068/edf719b2-353b-47e8-b7ba-a944c5f235f0

Video Link
[https://drive.google.com/file/d/1It1RlUXBqaED3alHofPCmtEgx089htop/view?usp=sharing](url)

@ChaudharyRaman  @prajjwalkapoor 

Please merge this pr and if in any problems are happening during integration do let me know.

### CODE REVIEW

**Comment by ChaudharyRaman** [comment]
@Divyanshusir please remove all the migrations file other than that this PR LGTM @prajjwalkapoor .

File: .gitignore [hunk 1]
```diff
@@ -72,3 +72,10 @@ node_modules/
 
 FusionIIIT/static/
 package-lock.json
+
+docker-entrypoint.sh
+
+migrations
```
Comment (ChaudharyRaman) [inline .gitignore]: just change this **migrations** line

**Comment by prajjwalkapoor** [comment]
LGTM. 
Just remove those migration files.

**Comment by Divyanshusir** [comment]
@ChaudharyRaman  I have removed all the migrations file . Please review it again and do let me know if there are any more issues.

**Comment by ChaudharyRaman** [comment]
LGTM Merging

**Pull request merged**